keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1083,6 +1105,38 @@ repeated evaluations if they are shared across validations.
     - expression: "itemMetadataNames.exists(name, name == 'required')"
```

Comment:
This example was added to make the basic idea clear, but we should think more about the details and improve it as necessary.

For example, using root CEL variables has a _significant_ impact on our ability to add root variables in the future. If a user names a variable `permissions` and we later add a `permissions` root variable, the variable names would collide. Alternative to consider: All user variables are fields of a `vars` (exact name TBD) variable in CEL to add scoping.

Reply:
Scoping is a concern, I agree. We could keep everything under something called:

* `celVars`
*  `composedExprs`
*  `k8s.vars`

I sort of like the idea of a root Kubernetes object.

Reply:
Maybe use the same identifier as the field name used in the resource to declare the variables? I.e. `variables`. That's basically what we do for params.

Reply:
(marking this as resolved, implemented the `variables` suggestion)